apriltag_ros/src/continuous_detector.cpp

```diff
@@ -86,6 +94,81 @@ bool ContinuousDetector::refreshParamsCallback(std_srvs::Empty::Request& req,
   return true;
 }
 
+void ContinuousDetector::publishEmptyDetection(const std_msgs::Header& header)
+{
+  AprilTagDetectionArray tag_detection_array;
+  tag_detection_array.header = header;
+  tag_detections_publisher_.publish(tag_detection_array);
+}
+
+cv::Point3f toCvPoint3f(const tf::Vector3& position)
+{
+  return cv::Point3f(position.getX(), position.getY(), position.getZ());
+}
+
+bool ContinuousDetector::isTagInFOV(const tf::Transform& tag_pose) const
+{
+  if (!camera_model_.initialized())
+  {
+    ROS_WARN_STREAM_THROTTLE(1.0, "Camera not initialized, skipping FOV check");
+    return false;
+  }
+
+  if (tag_pose.getOrigin().getZ() <= 0)
+  {
+    // handles Z = 0 (shouldn't happen) or Z < 0 (behind the camera)
+    return false;
+  }
+
+  // convert the 3d pose in the image frame to a 2d pixel frame
+  const auto image_pt = camera_model_.project3dToPixel(toCvPoint3f(tag_pose.getOrigin()));
+  ROS_DEBUG_STREAM_THROTTLE(1.0, 
+    "Frame " << camera_model_.cameraInfo().header.frame_id 
+      << " has target at pixel coordinate x: " << image_pt.x << ", y: " << image_pt.y);
+
+  // check if the image is within the target fov
+  return (fov_pixel_buffer_width_ <= image_pt.x 
+    && image_pt.x < (camera_model_.reducedResolution().width - fov_pixel_buffer_width_) 
+    && fov_pixel_buffer_height_ <= image_pt.y 
+    && image_pt.y < (camera_model_.reducedResolution().height - fov_pixel_buffer_height_));
```

Comment:
there was also an opportunity to do **const auto cam_resolution = camera_model_.reducedResolution();** here.

Reply:
Sorry missed this one. Will make another pr to cover it